### PR TITLE
Add CLI option to run all rules

### PR DESCRIPTION
**Pull request by jpsim**
Addresses #1170. /cc @norio-nomura

### CODE REVIEW

**Comment by norio-nomura** [comment]
Sorry, I failed to explain on the issue.
I expected that `OptInRule` will also be enabled by `--enable-all-rules` option.

File: Source/SwiftLintFramework/Models/Configuration.swift [hunk 1]
```diff
@@ -98,8 +99,10 @@ public struct Configuration: Equatable {
         // set the config threshold to the threshold provided in the config file
         self.warningThreshold = warningThreshold
 
-        // white_list rules take precendence over all else.
-        if !whitelistRules.isEmpty {
+        // Precedence is enableAllRules > whitelistRules > everything else
+        if enableAllRules {
+            rules = configuredRules
```
Comment (norio-nomura) [inline Source/SwiftLintFramework/Models/Configuration.swift]: Do `configuredRules` contain rules that is `OptInRule`?

**Comment by norio-nomura** [comment]
> I expected that OptInRule will also be enabled by --enable-all-rules option.

It seems `OptInRule` is not enabled by `--enable-all-rules`
```terminal.session
➜  12:57:34 $ cat .swiftlint-opt-in.yml
opt_in_rules:
 - attributes
➜  12:57:40 $ time swiftlint lint --benchmark --enable-all-rules --config .swiftlint-opt-in.yml >/dev/null 2>&1
swiftlint lint --benchmark --enable-all-rules --config .swiftlint-opt-in.yml   24.95s user 1.82s system 111% cpu 24.091 total
➜  12:58:13 $ wc -l benchmark_rules_2017_01_12_12_58_13.txt
      56 benchmark_rules_2017_01_12_12_58_13.txt
➜  12:58:37 $ time swiftlint lint --benchmark --enable-all-rules >/dev/null 2>&1
swiftlint lint --benchmark --enable-all-rules > /dev/null 2>&1  48.75s user 2.30s system 168% cpu 30.282 total
➜  12:59:27 $ wc -l benchmark_rules_2017_01_12_12_59_26.txt
      55 benchmark_rules_2017_01_12_12_59_26.txt
```

**Comment by jpsim** [comment]
@norio-nomura I think the issue is that the cache gets re-used, which takes the results from the previous run with fewer rules.

**Comment by jpsim** [comment]
Oops, no the issue was that the value from the CLI wasn't being passed to the Configuration initializer. Fixed now.

**Pull request merged**